Client/Frontend/Home/RecentTabsPanel.swift

```diff
@@ -0,0 +1,162 @@
+
+        tableViewController.profile = self.profile
+        tableViewController.homePanelDelegate = homePanelDelegate
+        tableViewController.historyPanel = historyPanel
+
+        self.addChildViewController(tableViewController)
+        self.view.addSubview(tableViewController.view)
+        self.view.addSubview(historyBackButton)
+        self.view.addSubview(recentlyClosedHeader)
+
+        historyBackButton.snp_makeConstraints { make in
+            make.top.left.right.equalTo(self.view)
+            make.height.equalTo(50)
+            make.bottom.equalTo(recentlyClosedHeader.snp_top)
+        }
+
+        recentlyClosedHeader.snp_makeConstraints { make in
+            make.top.equalTo(historyBackButton.snp_bottom)
+            make.height.equalTo(20)
+            make.bottom.equalTo(tableViewController.view.snp_top).offset(-10)
+            make.left.right.equalTo(self.view)
+        }
+
+        tableViewController.view.snp_makeConstraints { make in
+            make.left.right.bottom.equalTo(self.view)
+        }
+
+        tableViewController.didMoveToParentViewController(self)
+    }
+
+    @objc private func historyBackButtonWasTapped(gestureRecognizer: UITapGestureRecognizer) {
+        self.navigationController?.popViewControllerAnimated(true)
+    }
+}
+
+class RecentTabsPanelSiteTableViewController: SiteTableViewController {
+    weak var homePanelDelegate: HomePanelDelegate?
+    weak var historyPanel: HistoryPanel?
+    var recentlyClosedTabs: [String:[String]]!
+
+    init() {
+        super.init(nibName: nil, bundle: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(HistoryPanelSiteTableViewController.notificationReceived(_:)), name: NotificationPrivateDataClearedHistory, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(HistoryPanelSiteTableViewController.notificationReceived(_:)), name: NotificationDynamicFontChanged, object: nil)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.accessibilityIdentifier = "Recently Closed Tabs List"
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationPrivateDataClearedHistory, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationDynamicFontChanged, object: nil)
+    }
+
+    func notificationReceived(notification: NSNotification) {
+        switch notification.name {
+        case NotificationPrivateDataClearedHistory:
+            profile.prefs.setObject(nil, forKey: "recentlyClosedTabs")
+            break
+        case NotificationDynamicFontChanged:
+            reloadData()
+            break
+        default:
+            // no need to do anything at all
+            log.warning("Received unexpected notification \(notification.name)")
```

Comment:
Add an `assertionFailure()` here so this will crash on developer builds. That way, we can find it and fix it if it ever happens.